elasticache-redis.tf

```diff
@@ -0,0 +1,131 @@
+variable "redis_name" {
+  type        = "string"
+  default     = "redis"
+  description = "Redis name"
+}
+
+variable "redis_instance_type" {
+  type        = "string"
+  default     = "cache.t2.medium"
+  description = "EC2 instance type for Redis cluster"
+}
+
+variable "redis_cluster_size" {
+  type        = "string"
+  default     = "2"
+  description = "Redis cluster size"
+}
+
+variable "redis_cluster_enabled" {
+  type        = "string"
+  default     = "true"
+  description = "Set to false to prevent the module from creating any resources"
+}
+
+variable "redis_auth_token" {
+  type        = "string"
+  default     = ""
+  description = "Auth token for password protecting redis, transit_encryption_enabled must be set to 'true'! Password must be longer than 16 chars"
+}
+
+variable "redis_engine_version" {
+  type        = "string"
+  default     = "3.2.6"
+  description = "Version of Redis engine"
+}
+
+variable "redis_transit_encryption_enabled" {
+  type        = "string"
+  default     = "true"
+  description = "Enable TLS"
+}
+
+variable "redis_at_rest_encryption_enabled" {
+  type        = "string"
+  default     = "true"
+  description = "Enable Redis encryption at rest"
+}
+
+variable "redis_params" {
+  type        = "list"
+  default     = []
+  description = "A list of Redis parameters to apply. Note that parameters may differ from a Redis family to another"
+}
+
+variable "redis_maintenance_window" {
+  type        = "string"
+  default     = "sun:03:00-sun:04:00"
+  description = "Weekly time range during which system maintenance can occur, in UTC"
+}
+
+variable "redis_automatic_failover" {
+  type        = "string"
+  default     = "true"
+  description = "Whether to enable automatic_failover"
+}
+
+variable "redis_apply_immediately" {
+  type        = "string"
+  default     = "true"
+  description = "Whether to apply changes immediately or during the next maintenance_window"
+}
+
+locals {
+  redis_family     = "${format("redis%s", join(".", slice(split(".", var.redis_engine_version),0,2)))}"
+  redis_auth_token = "${length(var.redis_auth_token) > 0 ? var.redis_auth_token : join("", random_string.redis_auth_token.*.result)}"
+}
```

Comment:
redis_cluster_enabled = "${var.redis_cluster_enabled == "true"}"

Reply:
In the case of this module, I'm not sure if this really buys us much. I want to be able to toggle each service enabled/disabled individually if pass an override flag and if not, fall back on `var.enabled`. Given we need to pass in `var.enabled` (string) and not `local.enabled` (bool) into each submodule `local.enabled` is only used in resources within this meta module.

Reply:
In this case `local.redis_cluster_enabled` is a string. @joshmyers is not using a `boolean` (which is how we have been defining the *local* versions of enabled/disabled. We can fix later as this is working.